build/parser.rs

```diff
@@ -806,23 +861,11 @@ impl MavType {
             Float => quote! {#val = #buf.get_f32_le();},
             Double => quote! {#val = #buf.get_f64_le();},
             Array(t, size) => {
-                if size > 32 {
-                    // it is a vector
-                    let r = t.rust_reader(&quote!(let val), buf);
-                    quote! {
-                        for _ in 0..#size {
-                            #r
-                            #val.push(val).unwrap();
-                        }
-                    }
-                } else {
-                    // handle as a slice
-                    let r = t.rust_reader(&quote!(let val), buf);
-                    quote! {
-                        for idx in 0..#size {
-                            #r
-                            #val[idx] = val;
-                        }
+                let r = t.rust_reader(&quote!(let val), buf);
+                quote! {
+                    for idx in 0..#size {
```

Comment:
```suggestion
                    for v in &mut val {
```

(or something similar)